frontend/test/metabase/scenarios/visualizations/pie_chart.cy.spec.js

```diff
@@ -0,0 +1,52 @@
+import { restore, visitQuestionAdhoc } from "__support__/cypress";
+import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATASET;
+
+const testQuery = {
+  type: "query",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+    breakout: [["field-id", PRODUCTS.CATEGORY]],
+  },
+  database: 1,
+};
+
+describe("scenarios > visualizations > pie chart", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  function ensurePieChartRendered(rows, totalValue) {
+    cy.get(".Visualization").within(() => {
+      // detail
+      cy.findByText("Total").should("be.visible");
+      cy.findByTestId("detail-value").should("have.text", totalValue);
+
+      // slices
+      cy.findAllByTestId("slice").should("have.length", rows.length);
```

Comment:
We haven't used `testId` so far, but I guess this is a good place to start. I wrote one repro around Pie Chart just yesterday and had to use something like 
```javascript
cy.get("[class*=PieChart__Donut]")
  .find("path")
  .first()
```

@paulrosenzweig can you remember why we didn't use testIds so far? I can remember one of my first attempts to add it (it was maybe a unit test) and was discouraged but I can't remember the reason.

Reply:
In my personal opinion, it is better to use `testId` over querying using css classes because using css classes is tied to our approach of styling the app. In this particular case, it is a deprecated approach that uses `css-modules`. What do you think @nemanjaglumac @paulrosenzweig?

Reply:
> I absolutely agree. We're relying too much on css classes.

+1 on that. We initially didn't use them to avoid cluttering the app code with a testing concern. Now that Cypress use is more established, I think decoupling styling from testing is a bigger win.

It might not be worth chasing down all classes and replacing them since often they're equivalent. For example, it seems fine to [target icons that way](https://github.com/metabase/metabase/blob/master/frontend/test/__support__/commands.js#L34-L36). There, the class/icon name is equivalent to using a test id. If the class is removed down the line, switching it to an actual test id would be quick.

Reply:
@nemanjaglumac [this](https://github.com/metabase/metabase/pull/12919#issuecomment-659441019) was probably the comment you were thinking of. Maybe icons are the _only_ example I can think of where classes seem ok 😆

Reply:
I really know how to pick, don't I? :D
Thanks for the refresher, Paul. And thanks for sharing your thoughts here. I'm glad we're all on the same page.